ludwig/datasets/kaggle.py

```diff
@@ -40,8 +41,12 @@ def download_kaggle_dataset(
         api = create_kaggle_client()
         api.authenticate()
     with upload_output_directory(download_directory) as (tmpdir, _):
-        if kaggle_competition:
-            api.competition_download_files(kaggle_competition, path=tmpdir)
+        dataset_or_competition = kaggle_competition or kaggle_dataset_id
+        if filenames:
+            download_fn = api.competition_download_file if kaggle_competition else api.dataset_download_file
```

Comment:
This causes test_titanic_workflow to fail because it is mocking `competition_download_files`, but now we're calling `competition_download_file` instead.

See `test_titanic_workflow.py:88`:
`mock_kaggle_api.competition_download_files = download_files`